### PR TITLE
Add power control functions 

### PR DESCRIPTION
**Pull request by Bill2462**
I've added power control functions : 
power_down() - Enables soft power down mode. 
power_up() - Disables soft power down mode. 

Note !!! :
Calling any other function that uses Command Register will disable power down mode !!!

### CODE REVIEW

**Comment by Rotzbua** [comment]
Hi Bill2462, is your commit based on the discussion in #269 ?

File: src/MFRC522.cpp [hunk 1]
```diff
@@ -364,6 +364,38 @@ bool MFRC522::PCD_PerformSelfTest() {
 } // End PCD_PerformSelfTest()
 
 /////////////////////////////////////////////////////////////////////////////////////
+// Power control
+/////////////////////////////////////////////////////////////////////////////////////
+
+//IMPORTANT NOTE!!!!
+//Calling any other function that uses CommandReg will disable soft power down mode !!!
+
```
Comment (Rotzbua) [inline src/MFRC522.cpp]: Please add a reference to the data sheet, so later ppl can look it up.

File: src/MFRC522.cpp [hunk 2]
```diff
@@ -364,6 +364,38 @@ bool MFRC522::PCD_PerformSelfTest() {
...
+void MFRC522::power_down()//Note : Only soft power down mode is available throught software
```
Comment (Rotzbua) [inline src/MFRC522.cpp]: CS (code style): please like the other functions: `void MFRC522::power_down() {`

File: src/MFRC522.cpp [hunk 3]
```diff
@@ -364,6 +364,38 @@ bool MFRC522::PCD_PerformSelfTest() {
...
+{
```
Comment (Rotzbua) [inline src/MFRC522.cpp]: CS: see above

File: src/MFRC522.cpp [hunk 4]
```diff
@@ -364,6 +364,38 @@ bool MFRC522::PCD_PerformSelfTest() {
...
+	byte val=PCD_ReadRegister(CommandReg); // Read state of the command register
```
Comment (Rotzbua) [inline src/MFRC522.cpp]: CS: please add the spaces around the `=`
should be `byte val = PCD_ReadRegister(CommandReg); // Read state of the command register `

File: src/MFRC522.cpp [hunk 5]
```diff
@@ -364,6 +364,38 @@ bool MFRC522::PCD_PerformSelfTest() {
...
+	PCD_WriteRegister(CommandReg, val);//write new value to the command register
+}
+
+void MFRC522::power_up()
+{
+	byte val=PCD_ReadRegister(CommandReg); // Read state of the command register 
+	val &= ~(1<<4);// set PowerDown bit ( bit 4 ) to 0 
+	PCD_WriteRegister(CommandReg, val);//write new value to the command register
+	// wait until PowerDown bit is cleared (this indicates end of wake up procedure) 
+	unsigned long timer=millis();// create timer for timeout (just in case)
```
Comment (Rotzbua) [inline src/MFRC522.cpp]: `uint64_t timer = millis(); // create timer for timeout (just in case) `

File: src/MFRC522.h [hunk 6]
```diff
@@ -360,6 +360,12 @@ class MFRC522 {
 	bool PCD_PerformSelfTest();
 	
 	/////////////////////////////////////////////////////////////////////////////////////
+	// Power control functions
+	/////////////////////////////////////////////////////////////////////////////////////
+	void power_down();
```
Comment (Rotzbua) [inline src/MFRC522.h]: Naming like the other functions and more clear. I suggest `void PCD_SoftPowerDown();`

File: src/MFRC522.h [hunk 7]
```diff
@@ -360,6 +360,12 @@ class MFRC522 {
...
+	void power_up();
```
Comment (Rotzbua) [inline src/MFRC522.h]: `void PCD_SoftPowerUp();`

**Comment by Bill2462** [comment]
> Hi Bill2462, is your commit based on the discussion in #269 ?
No, it was written based on a Datasheet page 33 (8.6) and my own tests. 

I've tested it with DumpInfo example and I measured power consumption to be around 1.7 mA in soft power down mode. But there is an LED on my RFID board so it is probably less than a 1 mA. Power consumption during normal operation is around 17 mA.

Also, everything works fine after disabling power down mode. 
(As expected because according to the datasheet, in soft power down mode all register values, the FIFO buffer content and the
configuration keep their current contents.)

File: src/MFRC522.cpp [hunk 8]
```diff
@@ -364,6 +364,35 @@ bool MFRC522::PCD_PerformSelfTest() {
...
+}
+
+void MFRC522::PCD_SoftPowerUp(){
+	byte val = PCD_ReadRegister(CommandReg); // Read state of the command register 
+	val &= ~(1<<4);// set PowerDown bit ( bit 4 ) to 0 
+	PCD_WriteRegister(CommandReg, val);//write new value to the command register
+	// wait until PowerDown bit is cleared (this indicates end of wake up procedure) 
+	unsigned long timer = millis();// create timer for timeout (just in case) 
+	
+	while(millis()-500<=timer){ // set timeout to 500 ms
```
Comment (Rotzbua) [inline src/MFRC522.cpp]: `while(millis()<=timeout){ // set timeout to 500 ms `

**Pull request merged**